packages/rs-drive-abci/src/execution/engine/run_block_proposal/mod.rs

```diff
@@ -82,11 +85,31 @@ Your software version: {}, latest supported protocol version: {}."#,
 
             next_platform_version
         } else {
-            // Stay on the last committed platform version
+            // Stay on the last committed plat version
             last_committed_platform_version
         };
 
-        match platform_version
+        // Create a bock state from previous committed state
+        let mut block_platform_state = platform_state.clone();
+
+        // Set current protocol version to the block platform state
+        block_platform_state
+            .set_current_protocol_version_in_consensus(block_platform_version.protocol_version);
+
+        // Patch platform version and run migrations
+        // It modifies the protocol version to function version mapping to apply hotfixes
+        // Also it performs migrations to fix corrupted state or prepare it for new features
+        let block_platform_version = if let Some(patched_platform_version) = self.patch_platform(
+            block_proposal.height,
+            &mut block_platform_state,
+            transaction,
+        )? {
+            patched_platform_version
+        } else {
+            block_platform_version
+        };
```

Comment:
I would do an if needed { do things }

Reply:
We can't do `if`, because we need to run some logic anyway (clean up patches). Also, we would need to acquire a read lock twice. I renamed it and updated the comment.